### PR TITLE
wic wks and source plugin 

### PR DESCRIPTION
**Pull request by cheeyanglee**
goals: 
- provide another method to manage efi through new wks file and wic source plugin
- retire acrn-efi-setup which is a trick and required first bootup to setup boot entries
- retire .wic.acrn conversion which create redundant image and consume lots of space

### CODE REVIEW

**Comment by cheeyanglee** [comment]
see https://github.com/intel/meta-acrn/issues/16 for related discussion

File: docs/getting-started.md [hunk 1]
```diff
@@ -31,6 +31,10 @@ IMAGE_INSTALL_append_pn-acrn-image-sato = " core-image-weston-package"
 
 # set preferred kernel for sos
 PREFERRED_PROVIDER_virtual/kernel = "linux-intel-acrn-sos"
+
+# set kickstart file that affect sos image only
+WKS_FILE_pn-acrn-image-base = "acrn-bootdisk-microcode.wks.in"
+
```
Comment (rossburton) [inline docs/getting-started.md]: All the `acrn-image-*` recipes are SOS images, so can't this be set in the recipes (like they used to `inherit image-acrn`)

File: scripts/lib/wic/plugins/source/acrn-bootimg-efi.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
+#
```
Comment (rossburton) [inline scripts/lib/wic/plugins/source/acrn-bootimg-efi.py]: My only concern here is that this code is more than just a slim customisation layer, can it not inherit from a class in oe-core or is some refactoring planned?

Comment (cheeyanglee) [reply scripts/lib/wic/plugins/source/acrn-bootimg-efi.py]: refactoring needed once it comes to support features like pre-launched VM

File: docs/getting-started.md (hunk 1 above)
Comment (cheeyanglee) [reply docs/getting-started.md]: i tried it, weak assign in recipes not able to override the value assigned in meta-intel. 
hardcode in recipe will not allow local.conf to make further changes.

Comment (saininav) [reply docs/getting-started.md]: You can send the patch to meta-intel having below for intel-corei7-64 machine which will alow override. 

WKS_FILE ??= "${@bb.utils.contains_any("EFI_PROVIDER", "systemd-boot", "systemd-bootdisk-microcode.wks.in", "grub-bootdisk-microcode.wks.in", d)}"

File location:
http://git.yoctoproject.org/cgit/cgit.cgi/meta-intel/tree/conf/machine/intel-corei7-64.conf

Comment (cheeyanglee) [reply docs/getting-started.md]: there is a default WKS_FILE with "??=" in oe-core, where setting "??=" in meta-intel not able to override it.
https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/classes/image_types_wic.bbclass#n11

Comment (cheeyanglee) [reply docs/getting-started.md]: set WKS_FILE in  local.conf were not able to override the WKS_FILE set in image recipe, but WKS_FILE_pn-<image name> in local.conf still works. i had made the changes in the new push.

File: conf/distro/acrn-demo-sos.conf [hunk 3]
```diff
@@ -11,3 +11,12 @@ LINUX_ACRN_APPEND ?= "hvlog=2M@0x1FE00000"
 LINUX_GVT_APPEND ?= "i915.enable_gvt=1 i915.nuclear_pageflip=1 i915.avail_planes_per_pipe=0x01010F i915.domain_plane_owners=0x011111110000 i915.domain_scaler_owner=0x011100"
 
 APPEND += "${LINUX_ACRN_APPEND} ${LINUX_GVT_APPEND}"
+
+GRUB_BUILDIN_append = " chain "
```
Comment (saininav) [inline conf/distro/acrn-demo-sos.conf]: Direct boot mode use GRUB, in that case, how this chain-loading will work. Could you please check that ?
https://projectacrn.github.io/latest/introduction/index.html#direct-boot-mode

Comment (cheeyanglee) [reply conf/distro/acrn-demo-sos.conf]: current state only support for De-privilege boot mode.
Direct boot mode required multiboot module in grub-efi.
in that case  chain module are not required but multiboot module and grub menu with two entries one to direct boot, another one boot to native

**Comment by moto-timo** [comment]
I’d like us to revisit this direction...

**Comment by moto-timo** [comment]
@cheeyanglee when you get a chance, please rebase this PR.

**Comment by cheeyanglee** [comment]
acrn-hypervisor is dropping de-privileged boot (acrn.efi). 
another idea to implement it has been in master-next branch, hence closed.
https://github.com/intel/meta-acrn/pull/223

**Pull request closed without merging**